packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -326,7 +326,11 @@ export function useSettingsForBlockElement(
 			const sides = Array.isArray( supports?.spacing?.[ key ] )
 				? supports?.spacing?.[ key ]
 				: supports?.spacing?.[ key ]?.sides;
-			if ( sides?.length ) {
+			// Check if spacing type actually exists before adding sides.
+			if (
+				sides?.length &&
+				typeof updatedSettings.spacing?.[ key ] !== 'undefined'
```

Comment:
Should this be a truthy check instead of not undefined?

Reply:
Ah, I thought about it but wasn't sure if that was just the way of exposing only top and bottom margin controls 😅 

I'll give it a try.

Reply:
I've updated it, but now no dimensions controls seem to be showing for any blocks in classic themes 🤔 which I don't think is the intended result.

Reply:
Oh wait, actually that might be a Twenty Twenty thing, because I'm seeing padding controls for some blocks in Twenty Twenty One.

Reply:
Hrm... most likely not the intended result!

Reply:
One way to test it could be to try adding `add_theme_support( 'appearance-tools' );` to the classic theme you're testing?

Reply:
Yeah, confirmed it's a Twenty Twenty issue. So I guess this is working as expected?

Reply:
Good to hear!